[SITE-4379] Fixes for D11 support  and PHP 8.4 compatibility

rkunjappan opened the pull request.

chx committed:
Drupal 11 and PHP 8.4

rkunjappan commented on src/Form/DomainMaskingConfigForm.php:
Fix for the fatal error "ArgumentCountError: Too few arguments to function Drupal\Core\Form\ConfigFormBase::__construct(), 1 passed in /code/web/modules/contrib/pantheon_domain_masking/src/Form/DomainMaskingConfigForm.php on line 26 and exactly 2 expected in Drupal\Core\Form\ConfigFormBase->__construct() (line 44 of /code/web/core/lib/Drupal/Core/Form/ConfigFormBase.php).

rkunjappan commented on src/Middleware/DomainMaskingMiddleware.php:
```diff
@@ -131,7 +131,7 @@ public function handle(Request $request, $type = 1, $catch = TRUE): Response {
    * @return boolean
    */
-  protected function isPlatformDomainRequest(Request $request = NULL) {
+  protected function isPlatformDomainRequest(?Request $request = NULL) {
```
Fix added for "Deprecated in PHP 8.4: Parameter $request (Symfony\Component\HttpFoundation\Request) is implicitly nullable via default value null"

rkunjappan committed:
Fixes for Drupal 11

rkunjappan committed:
Update Readme file

scottbuscemi commented:
* Supports Drupal 9.4-11
* New release will be 2.1.0

rkunjappan commented on src/Form/DomainMaskingConfigForm.php:
```diff
@@ -22,9 +23,10 @@ class DomainMaskingConfigForm extends ConfigFormBase {
    */
-  public function __construct(ConfigFactoryInterface $config_factory) {
-    parent::__construct($config_factory);
+  public function __construct(ConfigFactoryInterface $config_factory,TypedConfigManagerInterface $typed_config) {
```
Fix added for the fatal error in Drupal 11 "ArgumentCountError: Too few arguments to function Drupal\Core\Form\ConfigFormBase::__construct(), 1 passed in /code/web/modules/contrib/pantheon_domain_masking/src/Form/DomainMaskingConfigForm.php on line 26 and exactly 2 expected in Drupal\Core\Form\ConfigFormBase->__construct() (line 44 of /code/web/core/lib/Drupal/Core/Form/ConfigFormBase.php).

greg-1-anderson replied:
Does this change still work in Drupal 9.4+ and 10?

Also, to conform to PHP style guidelines, there should be a space after commas (above and below). Is phpcs not configured for this project?

pwtyler replied:
There is no linting or testing for this module 😭

rkunjappan replied:
@greg-1-anderson  The constructor signature change in ConfigFormBase was introduced in Drupal 10.2.
https://www.drupal.org/node/3404140
 I've tested the updated code on Drupal 9.4+ and 10.x without encountering any runtime errors or warnings. I can re-test this in a Drupal 9 production environment.
Eventhough it's working for me, do you see a better way to approach this ?

And I fixed the PHP style issue in the code.

rkunjappan replied:
@greg-1-anderson  I tested the changes for the module  on drupal 9 with multiple ways. There is no error in UI, log messages, PHPStan analysis, but if  test with drupal-check it will throw error for parameter count.
` php vendor/bin/drupal-check -d web/modules/contrib/pantheon_domain_masking`
 **" Method Drupal\Core\Form\ConfigFormBase::__construct() invoked with 2  
         parameters, 1 required."**

rkunjappan committed:
Resolve compatibility issue in form constructor

rkunjappan committed:
apply phpcs fixes to code formatting

rkunjappan commented on src/Form/DomainMaskingConfigForm.php:
```diff
@@ -26,7 +26,6 @@ class DomainMaskingConfigForm extends ConfigFormBase {
   public function __construct(ConfigFactoryInterface $config_factory,TypedConfigManagerInterface $typed_config) {
     parent::__construct($config_factory,$typed_config);
     $this->configFactory = $config_factory;
-    $this->typedConfigManager = $typed_config;
```
Removed the assignment of $typed_config to prevent PHPStan warning about undefined property in Drupal 9

rkunjappan committed:
Add linting to github action workflow

rkunjappan committed:
Apply PHPCS  fixes

rkunjappan commented on .github/workflows/ci.yml:
```diff
@@ -13,3 +13,28 @@ jobs:
         uses: pantheon-systems/phpcompatibility-action@v1
         with:
           test-versions: 7.4-
+  linting:
```
Added linting to the github action workflow. Do I need to make any change to the script in composer.json?
https://github.com/pantheon-systems/pantheon_domain_masking/blob/ac6f1bd62c4cfb0c37bb4f1ca4a40924f910aa6a/composer.json#L19

rkunjappan committed:
Update description in composer file

rkunjappan commented on .github/workflows/ci.yml:
```diff
@@ -13,3 +13,28 @@ jobs:
         uses: pantheon-systems/phpcompatibility-action@v1
         with:
           test-versions: 7.4-
```
The test-version for pantheon-systems/phpcompatibility-action@v1 is 7.4, it should be updated to php 8, right?

rkunjappan committed:
Add phpcbf script to github workflow

pwtyler commented on .github/workflows/ci.yml:
```diff
@@ -13,3 +13,30 @@ jobs:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/pantheon-public/build-tools-ci:8.x-php8.1
+      options: --user root
```
We should be able to run this on ubuntu-latest with `shivammathur/setup-php`, no need for build tools image.

pwtyler commented on .github/workflows/ci.yml:
```diff
@@ -13,3 +13,30 @@ jobs:
+      COMMIT_SHA: ${{ github.sha }}
+      SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
+      SANDBOX_SSH_KEY: ${{ secrets.SANDBOX_SSH_KEY }}
+      BASH_ENV: ~/.bashrc
```
These can be dropped from linting, not using terminus.

rkunjappan committed:
Add php lint check to ci workflow

rkunjappan committed:
Update docblock to describe constructor dependencies in DomainMaskingConfigForm

rkunjappan commented:
@greg-1-anderson @pwtyler 
Applied the changes you have mentioned and this is ready for review.

- Updated github workflow and composer.json to include PHP linting.
- Added comment in src/Form/DomainMaskingConfigForm.php about the new argument on  __construct()

pwtyler commented on .github/workflows/ci.yml:
```diff
@@ -3,13 +3,32 @@ on:
+      - name: Code sniff
+        run: composer run-script codesniff
+      - name: Code fix
+        run: composer run-script code:fix
```
fix won't help without a pushback. we've done this on a couple of projects, but I don't think we need that here today.

rkunjappan committed:
Add PHP 8.3 linting job using shivammathur/setup-php

rkunjappan committed:
Fix linting issue on ConFigForm Docblock

pwtyler commented on .github/workflows/ci.yml:
```diff
@@ -3,13 +3,24 @@ on:
-    name: PHP Compatibility
+
     steps:
       - name: Checkout
         uses: actions/checkout@v2
```
```suggestion
      - name: Checkout
        uses: actions/checkout@v4
```

pwtyler committed:
Update .github/workflows/ci.yml
